cluster_env.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+
+_setup_cluster_env() {
+    local overwrite_vars=false
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            --overwrite)
+                overwrite_vars=true
+                shift
+                ;;
+            *)
+                echo "Warning: Unknown argument '$1' ignored" >&2
+                shift
+                ;;
+        esac
+    done
+
+    if ! command -v jq &> /dev/null; then
+        echo "Error: jq is not installed or not in your PATH. Please install jq." >&2
+        return 1
+    fi
+
+    local CURRENT_HOSTNAME
+    CURRENT_HOSTNAME=$(hostname)
+    local CONFIG_FILE
+    CONFIG_FILE="$(dirname "${BASH_SOURCE[0]}")/clusters.json"
+
+    if [ ! -f "$CONFIG_FILE" ]; then
+        echo "Error: Configuration file '$CONFIG_FILE' not found." >&2
+        return 1
+    fi
+
+    local cluster_found=false
+    while IFS= read -r pattern; do
+        # Convert glob to regex: escape dots, replace * with .*
+        regex="${pattern//./\\.}"
+        regex="${regex//\*/.*}"
+        if [[ "$CURRENT_HOSTNAME" =~ ^$regex$ ]]; then
+            cluster_found=true
+            local cluster_name
+            cluster_name=$(jq -r --arg p "$pattern" '.[$p].name' "$CONFIG_FILE")
```

Comment:
Can you separate 1) the logic to detect the cluster 2) the block to set env variable?